hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeConfiguration.java

```diff
@@ -137,7 +146,7 @@ public class DatanodeConfiguration {
    * missed. With max threads 5, optimistically DN can handle 1500 individual
    * container delete tx in 60s with RocksDB cache miss.
    */
-  @Config(key = "block.delete.threads.max",
+  @Config(key = BLOCK_DELETE_THREAD_MAX,
```

Comment:
If we change a configuration's value, if only need to change the value itself, and no additional operations are performed.  we can just add a `reconfigurable = true` in `@Config` annotation. the reconfigurable configuration can be auto registered. 
```java
      reconfigurationHandler =
          new ReconfigurationHandler("DN", conf, this::checkAdminPrivilege)
              .register(HDDS_DATANODE_BLOCK_DELETE_THREAD_MAX,
                  this::reconfigBlockDeleteThreadMax)
              .register(OZONE_BLOCK_DELETING_SERVICE_WORKERS,
                  this::reconfigDeletingServiceWorkers)
              .register(conf.getObject(DatanodeConfiguration.class));  // register the reconfigurable configuration
```
then this  configuration's value can be modify in runtime by `ozone admin reconfig` command.

But If need to do something extra when modifying the configuration value, such as modifying the thread pool (`block.delete.threads.max`), we need to register the operation manually.

Reply:
So either register the property explicitly with a consumer handler, or set the  "reconfigurable = true"  attribute to the property.  The later way the property will be auto registered with a default consumer handler, right? @xichen01

Reply:
Yes, if you set the "reconfigurable = true" attribute, the configuration will be auto handled when "reconfigure".
but one point needs to notice:

The object registered to `ReconfigurationHandler` must be the same conf as the one that wants to get the value after reconfiguring.
such as:
```java

      DatanodeConfiguration dnConf1 = conf.getObject(DatanodeConfiguration.class)
      DatanodeConfiguration dnConf2 = conf.getObject(DatanodeConfiguration.class)

      reconfigurationHandler =
          new ReconfigurationHandler("DN", conf, this::checkAdminPrivilege)
              .register(dnConf1); 
```
After reconfiguring, only `dnConf1 ` can get the changed value, because only `dnConf1` was registered to the `ReconfigurationHandler`

Reply:
@xichen01 , thanks for the explain.